Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/function/AzureFunctionMvpModel.java

```
@@ -0,0 +1,154 @@
+
+    private AzureFunctionMvpModel() {
+        subscriptionIdToFunctionApps = new ConcurrentHashMap<>();
+    }
+
+    public static AzureFunctionMvpModel getInstance() {
+        return SingletonHolder.INSTANCE;
+    }
+
+    public FunctionApp getFunctionById(String sid, String id) throws IOException {
+        Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
+        FunctionApp app = azure.appServices().functionApps().getById(id);
+        if (app == null) {
+            throw new IOException(CANNOT_GET_WEB_APP_WITH_ID + id); // TODO: specify the type of exception.
+        }
+        return app;
+    }
+
+    public void deleteFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().deleteById(appId);
+        subscriptionIdToFunctionApps.remove(sid);
+    }
+
+    public void restartFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().getById(appId).restart();
+    }
+
+    public void startFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().getById(appId).start();
+    }
+
+    public void stopFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().getById(appId).stop();
+    }
+
+    /**
+     * List app service plan by subscription id and resource group name.
+     */
+    public List<AppServicePlan> listAppServicePlanBySubscriptionIdAndResourceGroupName(String sid, String group) {
+        List<AppServicePlan> appServicePlans = new ArrayList<>();
+        try {
+            Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
+            appServicePlans.addAll(azure.appServices().appServicePlans().listByResourceGroup(group));
```

are there any reasons of swallow exception?

---

Plan to swallow exception in case blocking UI thread, changed to throw exception in base util method and handle them in caller